edith-step/src/ReceiptInput.js

```diff
@@ -0,0 +1,157 @@
+import React from 'react';
+import axios from 'axios';
+
+export default class ReceiptInput extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {items: [], itemName: '', itemPrice: 0.0, itemQuantity: 1,
+                  itemCategory: 'category', itemReceiptId: 'receiptId' };
+    this.handleChange = this.handleChange.bind(this);
+    this.handleSubmit = this.handleSubmit.bind(this);
+    this.getDate = this.getDate.bind(this);
+  }
+<<<<<<< HEAD
+=======
+  
+  /**
+   * Calculates the current in yyyy-mm-dd format.
+   * @return the date in yyyy-mm-dd format
+   */
+  getDate() {
+      const date = new Date(Date.now());
+      let month = date.getMonth() + 1;
+      month < 10 ? month = "0" + month.toString() : month = month.toString();
+      let day = date.getDate();
+      day < 10 ? day = "0" + day.toString() : day = day.toString();
+      let year = date.getFullYear();
+      return [year, month, day].join("-");   
+  }
+>>>>>>> user-charts
+
+  handleSubmit(e) {
+    e.preventDefault();
+
+    if (this.state.itemName.length === 0) {
+      return;
+    }
+  
+    axios({
+      method: 'post',
+      url: '/user-stats-servlet',
+      data: {
+        itemName: this.state.itemName,
+        itemCategory: this.state.itemCategory,
+        itemPrice: this.state.itemPrice,
+        itemQuantity: this.state.itemQuantity,
+        itemDate: this.getDate(),
+        itemReceiptId: this.state.itemReceiptId
+      }
+    }).catch((err) => {
+      console.log(err);
+    });
+
+    const newItem = {
+      itemName: this.state.itemName,
+      itemPrice: this.state.itemPrice,
+      itemQuantity: this.state.itemQuantity,
+      id: Date.now()
+    };
+    
+    this.setState(state => ({
+      items: state.items.concat(newItem),
+      itemName: '',
+      itemPrice: 0.0,
+      itemQuantity: 1
+    }));
+<<<<<<< HEAD
+=======
+
+    axios({
+      method: 'post',
+      url: '/user-stats-servlet',
+      data: {
+        itemName: this.state.itemName,
+        itemPrice: this.state.itemPrice,
+        itemQuantity: this.state.itemQuantity,
+        itemDate: this.getDate()
+      }
+    }).then((response) => {
+      console.log(response);
```

Comment:
is this .then with a print statement necessary? could it be a catch instead if response isn't used

Reply:
I changed it.